recipes/parallelio/build.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+export NETCDF_C_PATH=$(dirname $(dirname $(which nc-config)))
+export NETCDF_FORTRAN_PATH=$(dirname $(dirname $(which nf-config)))
+
+mkdir build
+cd build
+FC=mpifort CC=mpicc CXX=mpicxx cmake \
+    -DCMAKE_INSTALL_PREFIX=$PREFIX \
+    -DBUILD_SHARED_LIBS:BOOL=ON \
+    -DPIO_USE_MALLOC:BOOL=ON \
+    -DPIO_ENABLE_TOOLS:BOOL=OFF \
+    -DPIO_ENABLE_TESTS:BOOL=OFF \
+    -DPIO_ENABLE_EXAMPLES:BOOL=OFF \
+    -DPIO_ENABLE_TIMING:BOOL=OFF \
+    -DNetCDF_C_PATH=$NETCDF_C_PATH \
+    -DNetCDF_Fortran_PATH=$NETCDF_FORTRAN_PATH ..
+
+make
+
+# make tests
+# ctest
```

Comment:
I had trouble with some tests when I ran locally:
```
[ 71%] Linking C executable piodecomptest
/home/xylar/miniconda3/conda-bld/parallelio_1661760167351/_build_env/bin/../lib/gcc/x86_64-conda-linux-gnu/12.1.0/../../../../x86_64-conda-linux-gnu/bin/ld: CMakeFiles/piodecomptest.dir/mpi_argp.c.o: in function `argp_usage':
mpi_argp.c:(.text.argp_usage+0x0): multiple definition of `argp_usage'; CMakeFiles/piodecomptest.dir/piodecomptest.c.o:piodecomptest.c:(.text.argp_usage+0x0): first defined here
/home/xylar/miniconda3/conda-bld/parallelio_1661760167351/_build_env/bin/../lib/gcc/x86_64-conda-linux-gnu/12.1.0/../../../../x86_64-conda-linux-gnu/bin/ld: CMakeFiles/piodecomptest.dir/mpi_argp.c.o: in function `_option_is_short':
mpi_argp.c:(.text._option_is_short+0x0): multiple definition of `_option_is_short'; CMakeFiles/piodecomptest.dir/piodecomptest.c.o:piodecomptest.c:(.text._option_is_short+0x0): first defined here
/home/xylar/miniconda3/conda-bld/parallelio_1661760167351/_build_env/bin/../lib/gcc/x86_64-conda-linux-gnu/12.1.0/../../../../x86_64-conda-linux-gnu/bin/ld: CMakeFiles/piodecomptest.dir/mpi_argp.c.o: in function `_option_is_end':
mpi_argp.c:(.text._option_is_end+0x0): multiple definition of `_option_is_end'; CMakeFiles/piodecomptest.dir/piodecomptest.c.o:piodecomptest.c:(.text._option_is_end+0x0): first defined here
```
I don't yet feel like I have the expertise to debug this but I hope to figure out whether I have something misconfigured here (most likely a missing cmake flag) or if there is a problem with the test.

Reply:
I've not seen this error before, but the problem appears to be in <argp.h>

Reply:
I have commented out that test and can get the rest to build but most (all?) MPI tests fail on run because I don't have enough cores on my laptop.  I think this is going to be even worse for CI.  So I think we should let `ctest` be for now.